Fix "(wrong-type-argument stringp default)" in url-compat-retrieve.
https://github.com/hexmode/mediawiki-el/issues/27

--- mediawiki-el-2.2.9-orig/mediawiki.el
+++ mediawiki-el-2.2.9/mediawiki.el
@@ -414,10 +414,15 @@
 POST-PROCESS is the function to call for post-processing.
 BUFFER is the buffer to store the result in.  CALLBACK will be
 called in BUFFER with CBARGS, if given."
-  (let ((url-user-agent (concat (string-trim (if (functionp url-user-agent)
-                                                 (funcall url-user-agent)
-                                               url-user-agent))
-                                " mediawiki.el " mediawiki-version "\r\n")))
+  (let ((url-user-agent
+	 (if (eq url-user-agent 'default)
+	     'default
+	   (concat (string-trim (if (functionp url-user-agent)
+				    (funcall url-user-agent)
+				  url-user-agent))
+		   " mediawiki.el " mediawiki-version "\r\n")))
+	(url-package-name "mediawiki.el")
+	(url-package-version mediawiki-version))
     (cond ((boundp 'url-be-asynchronous) ; Sniff w3 lib capability
            (if callback
                (setq url-be-asynchronous t)
